client/components/Assignments/AssignmentItem/fields/Headline.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+interface IProps {
+    assignment: any;
+    loadArchiveItem: (assignment) => any;
+}
+
+export const HeadlineComponent = ({assignment, loadArchiveItem}: IProps) => {
+    const itemIds = assignment.item_ids;
+
+    if (!itemIds) {
+        return null;
+    }
+
+    const itemId = assignment.item_ids[0];
+
+    if (!itemId) {
+        return null;
+    }
+
+    const [item, setItem] = React.useState(null);
```

Comment:
can you stick to class API?

Reply:
I'll probably just change this to accept the item as a prop instead of fetching it from here. But why do you prefer the class API? We already use function components, and hooks are the way to go to make them stateful if there's no reason to create a class

Reply:
See this discussion https://github.com/superdesk/superdesk-ui-framework/pull/436#discussion_r383839845

Hooks are a totally different way of dealing with state, which has a significant learning curve. It increases the number of things everyone needs to understand in order to maintain the codebase but doesn't bring many advantages. There may be one here and there, but for the most part, one can achieve the same things using classes and higher-order components.

I'm not saying that we should never use hooks, but until we bump into cases where hooks would have a significant impact, I'd say let's keep it simple and use the same thing in all places.

Reply:
FC are also a different way (compared to Classes) of creating components, but that's not a reason to use one only. To maintain the codebase you need to know React, and hooks are a more automatic and simpler (IMO) way of handling state, you do the same with less code.

Also, even tho they do the same as setState, they're not really the same because they allow extra functionality that can be very useful in the future. They allow us to create common hooks to modify state and even use other modules ([this one](https://github.com/zeit/swr) as an example) that simplify our code doing a lot of things automatically, without having to deal with classes hierarchy.

Anyway, I'm not gonna discuss it anymore cause I won't push any hooks to this PR, as I'm probably gonna replace this component with one that gets the data as a prop 👍

Reply:
> FC are also a different way (compared to Classes) of creating components, but that's not a reason to use one only.

Function components have almost no complexity or learning curve. You copy-paste the code from the render block of a class component into a function, update props usage and you have a function component. That being said, function components have limitations that class components don't have - usage or refs or lifecycle methods.

By sticking to one way of doing things we make the codebase easier for others to contribute/maintain. If we use only class components, no one has to read about the differences to function components and when to use which. That I think is a pretty good reason to use only one.

Now functional components aren't a big deal and there isn't _that_ much to learn, so I'm not complaining about both being used. It's different with hooks since there's much more everyone would need to learn.

> hooks are a more automatic and simpler (IMO) way of handling state, you do the same with less code.

What exactly do you find simpler? Regarding less code, I don't think that 5 lines of constructor code are a problem. We should focus more to keeping low complexity than saving a few lines of boilerplate.

I'm not saying hooks are bad or can't be helpful in the future. They can, but I suggest waiting for that future instead of spending resources upfront before we need it.

Do you have any specific ideas on improving something in superdesk that can't be done/is hard to do using higher-order components?

Don't take this personally, I'm just trying to look at the cost/benefit ratio, look at the objective arguments for/against and base the decisions on that.

Reply:
HOC can also be made with functional components, I'm ok with that.